classes/class-wc-connect-stripe.php

```diff
@@ -285,7 +285,8 @@ public function connection_success_notice() {
 		 */
 		public function show_connected_account( $settings ) {
 			ob_start();
-			do_action( 'enqueue_wc_connect_script', 'wc-connect-stripe-connect-account' );
```

Comment:
I don't get why all this PR is needed. I think changing this line with:
```php
			$screen = get_current_screen();
 			if ( // WooCommerce » Settings » Payments » Stripe.
				'woocommerce_page_wc-settings' === $screen->base
				&& isset( $_GET['tab'] ) && 'checkout' === $_GET['tab']
				&& isset( $_GET['section'] ) && 'stripe' === $_GET['section']
			) {
				do_action( 'enqueue_wc_connect_script', 'wc-connect-stripe-connect-account' );
			}
```
Should fix the problem. Unless I'm missing something obvious?

Reply:
I was going to test this to make sure but didn't get around to it: my understanding was that `show_connected_account` was run on a hook that was too early for `get_current_screen`, which returns `null` until screen is initialized (before `current_screen` hook [[doc](https://codex.wordpress.org/Function_Reference/get_current_screen)]).

That said, I believe it can still be done by checking for `'wc-settings' === $_GET['page']`.

Reply:
The problem for me is that you're changing the logic to be more complex (hooks upon callbacks upon more hooks). If changing this line to add the conditional that checks for `'wc-settings' === $_GET['page']` is enough to solve this issue, we should change just that. Do you agree, or do you see any problem relying on the `$_GET['page']` variable?

Reply:
I don't see a problem – I must have just been thinking "if there is a `get_current_screen` function, it should be used". I've updated it with that change.